### ISSUE TITLE
Mask-RCNN: Mask target resizing fails with invalid resize

### ISSUE THREAD

**Issue by dsuess** [open-mmlab/mmdetection#1687]
**Describe the bug**

While training a Mask-RCNN on a custom dataset, I get the following error:

```
cv2.error: OpenCV(4.1.1) /io/opencv/modules/imgproc/src/resize.cpp:3720: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
```

This is raised from [mask_target_single](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/core/mask/mask_target.py#L31) and is due to one of the bounding boxes in `pos_proposals` having a coordinate `-1`.

A similar error at a different position was raised #370 for bounding-box datasets, but since the bounding boxes for instance segmentation datasets are computed on-the-fly, it's not fixed by it.

**Reproduction**
1. What command or script did you run?
Standard training script.
2. Did you make any modifications on the code or config? Did you understand what you have modified?
No. 
3. What dataset did you use?
Proprietary, but I think any dataset with instance masks close enough to the boundary + some rotation augmentation should replicate this.

**Error traceback**
If applicable, paste the error trackback here.
```
...
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmcv/runner/runner.py", line 358, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmcv/runner/runner.py", line 264, in train
    self.model, data_batch, train_mode=True, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/apis/train.py", line 38, in batch_processor
    losses = model(**data)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 152, in forward
    outputs = self.parallel_apply(replicas, inputs, kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 162, in parallel_apply
    return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/parallel/parallel_apply.py", line 83, in parallel_apply
    raise output
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/parallel/parallel_apply.py", line 59, in _worker
    output = module(*input, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/core/fp16/decorators.py", line 49, in new_func
    return old_func(*args, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/models/detectors/base.py", line 117, in forward
    return self.forward_train(img, img_meta, **kwargs)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/models/detectors/two_stage.py", line 254, in forward_train
    self.train_cfg.rcnn)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/models/mask_heads/fcn_mask_head.py", line 111, in get_target
    gt_masks, rcnn_train_cfg)
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/core/mask/mask_target.py", line 12, in mask_target
    mask_targets = torch.cat(list(mask_targets))
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmdet/core/mask/mask_target.py", line 32, in mask_target_single
    mask_size[::-1])
  File "/home/users/daniel/.local/conda/envs/mmpond/lib/python3.6/site-packages/mmcv/image/transforms/resize.py", line 45, in imresize
    img, size, interpolation=interp_codes[interpolation])
cv2.error: OpenCV(4.1.1) /io/opencv/modules/imgproc/src/resize.cpp:3720: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
```

**Bug fix**
We should probably just clip the `proposal_np` to valid values. I'll create a PR.
